### ISSUE TITLE
socket.http PORT field does nothing

### ISSUE THREAD

**Issue by daurnimator** [diegonehab/luasocket#160]
The socket.http PORT field does nothing.
Instead the default is brought in via https://github.com/diegonehab/luasocket/blob/46d7e75f3e71d75bf07ae83d7df4aa276e484473/src/http.lua#L253
e.g. try this:
```lua
local http = require "socket.http"
http.PORT = 1234;
print(http.request("http://httpbin.org/get"))
```
It goes ahead and makes the request to port 80 anyway.

As this hasn't worked for a while, I suggest just removing PORT from the source code + docs.

**Comment by diegonehab** [comment]
This is a bug, right? I think it is a bug in adjustrequest().

**Comment by daurnimator** [comment]
> This is a bug, right? I think it is a bug in adjustrequest().

Well it's a difference between documented behaviour and actual behaviour.

My preference here is to change the docs rather than the code, as it has been broken for a long time, and I doubt anyone is using the feature.

**Comment by diegonehab** [comment]
I agree that modifying the field (if it worked) would be tricky. Perhaps it is really best to remove it from the documentation and keep it as a literal local constant in http.lua. And also fix the code that currently overrides it. :)

**Comment by daurnimator** [comment]
Fixed with 5b4b9158799293eeaf8439d40a7845b5b0a5e125

>also fix the code that currently overrides it.

Do you know of any code that does?

**Comment by diegonehab** [comment]
Was this here:

    if nreqt.port == "" then nreqt.port = 80 end
    socket.try(nreqt.host and nreqt.host ~= "", 
        "invalid host '" .. base.tostring(nreqt.host) .. "'")
